lib/fluent/plugin/out_lm.rb

```diff
@@ -105,11 +105,11 @@ def process_record(tag, time, record)
           lm_event["_lm.resourceId"] = record["_lm.resourceId"]
         end
       else
-        if record["service"]==nil
+        if record["resource.service.name"]==nil
           log.error "When device_less_logs is set \'true\', record must have \'service\'. Ignoring this event #{lm_event}."
           return nil
         else
-          lm_event["service"] = encode_if_necessary(record["service"]) 
+          lm_event["resource.service.name"] = encode_if_necessary(record["resource.service.name"]) 
         end  
```

Comment:
looks fine. Just a small question : are we going to use namespace (`resource.namespace.name`) something like this field for anomaly detection ? If yes, we should add another if check, if we have that field, populate it whenever `device_less_logs` is set to `true`.

Reply:
Yes, we will use `resource.namespace.name` for anomaly detection but it is not mandatory `resource.service.name` is.